src/store/GlobalState.js

```diff
@@ -0,0 +1,155 @@
+import { createContext, useEffect, useReducer } from 'react';
+import { collection, doc, addDoc, deleteDoc, updateDoc, query, onSnapshot } from 'firebase/firestore';
+
+import db from '../firebase';
+import appReducer from './appReducer';
+import { getCurrentMonth, getUKFormattedDate } from '../utils/helpers';
+
+const currentMonth = getCurrentMonth();
+
+const initialState = {
+  expenses: [],
+  selectedMonthExpenses: [],
+  currentMonth: currentMonth,
+  selectedMonth: currentMonth,
+  lastAddedExpense: {},
+  expenseToDelete: {},
+  expenseToEdit: {},
+  isConfirmMonthModalOpen: false,
+  isConfirmDeleteModalOpen: false,
+  isEditExpenseModalOpen: false,
+  didErrorOccur: false,
+  isAlertOpen: false,
+  userAction: '',
+};
+
+export const GlobalContext = createContext(initialState);
+
+export const GlobalProvider = ({ children }) => {
```

Comment:
the name `global` is not very speaking and smells funny to me. consider, we later want to add another context for a more specific purpose. is there a better name for this? can we maybe even split the contexts to be more specific and achieve better separation of concerns?